dotCMS/src/main/java/com/dotcms/rest/api/v1/experiment/ExperimentFactory.java

```diff
@@ -0,0 +1,233 @@
+        } catch (DotDataException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static List<Rule> getRuleByExperiment(final Experiment experiment) {
+        try {
+            return new DotConnect()
+                    .setSQL(GET_RULES_BY_EXPERIMENT_QUERY)
+                    .addParam(experiment.getKey())
+                    .loadObjectResults()
+                    .stream()
+                    .map(result -> result.get("rule_id").toString())
+                    .map(ruleID -> {
+                        try {
+                            return APILocator.getRulesAPI()
+                                    .getRuleById(ruleID, APILocator.systemUser(), false);
+                        } catch (DotDataException | DotSecurityException e) {
+                            return null;
+                        }
+                    })
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList());
+        } catch (DotDataException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static VariantsCollection getVariantByExperiment(final Experiment experiment,
+            final String pageInode) {
+        try {
+            final IHTMLPage page = APILocator.getHTMLPageAssetAPI()
+                    .findPage(pageInode, APILocator.systemUser(), false);
+
+            final List<ExperimentVariant> variants = new DotConnect()
+                    .setSQL(GET_VARIANT_BY_EXPERIMENT_QUERY)
+                    .addParam(experiment.getKey())
+                    .loadObjectResults()
+                    .stream()
+                    .map(map -> new ExperimentVariant(
+                            new DotCMSVariant(map.get("name").toString(),
+                                    map.get("domain").toString(), map.get("key").toString()),
+                            Integer.parseInt(map.get("trafficPercentage").toString()),
+                            Boolean.valueOf(map.get("original").toString()))
+                    )
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList());
+
+            return new VariantsCollection(variants, page);
+        } catch (DotDataException | DotSecurityException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static Optional<Experiment> getExperiment(final String experimentValue) {
+        try {
+            return new DotConnect()
+                    .setSQL(GET_EXPERIMENT_BY_KEY_QUERY)
+                    .addParam(experimentValue)
+                    .loadObjectResults()
+                    .stream()
+                    .map(experimentMap -> createExperiment(experimentMap))
+                    .findFirst();
+        } catch (DotDataException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @WrapInTransaction
+    public static void save(final ExperimentForm experimentForm) throws DotDataException {
```

Comment:
I think forms shouldn't reach this level. It should take "Experiment" instead of "ExperimentForm"